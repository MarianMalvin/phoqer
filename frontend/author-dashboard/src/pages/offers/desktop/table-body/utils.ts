import { TableCellSizeEnum } from 'phoqer';

export interface HeaderType {
    key: string;
    title: string;
}

export const headerOrder: HeaderType[] = [
    { key: 'status', title: 'Status' },
    { key: 'options', title: 'Options' },
    { key: 'images', title: 'Images' },
    { key: 'title', title: 'Title' },
    { key: 'category', title: 'Category' },
    { key: 'reviews', title: 'Reviews' },
    { key: 'price', title: 'Price' },
    { key: 'sale_percentage', title: 'Sale percentage' },
    { key: 'sale_description', title: 'Sale description' },
    { key: 'description', title: 'Description' },
];

export const sizeMap: { [key: string]: TableCellSizeEnum } = {
    status: TableCellSizeEnum.Small,
    options: TableCellSizeEnum.Middle,
    images: TableCellSizeEnum.Small2Extra,
    title: TableCellSizeEnum.Large5Extra,
    description: TableCellSizeEnum.Large8Extra,
    sale_percentage: TableCellSizeEnum.MiddleExtra,
    sale_description: TableCellSizeEnum.Large4Extra,
    reviews: TableCellSizeEnum.Middle,
    price: TableCellSizeEnum.Middle,
    category: TableCellSizeEnum.Middle,
};
