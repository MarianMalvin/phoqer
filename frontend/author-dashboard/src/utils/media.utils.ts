export const isTablet = window.innerWidth > 768;
