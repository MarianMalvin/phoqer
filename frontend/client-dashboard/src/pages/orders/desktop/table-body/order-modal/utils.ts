export const cellMap = [
    { key: 'title', title: 'Title' },
    { key: 'status', title: 'Status' },
    { key: 'date', title: 'Created' },
    { key: 'expired', title: 'Expired' },
    { key: 'price', title: 'Price' },
    { key: 'category', title: 'Category' },
    { key: 'country', title: 'Country' },
    { key: 'city', title: 'City' },
    { key: 'zip', title: 'ZIP' },
    { key: 'address', title: 'Address' },
    { key: 'comment', title: 'Comment' },
    { key: 'sale', title: 'Sale' },
    { key: 'sale_description', title: 'Sale Description' },
];
