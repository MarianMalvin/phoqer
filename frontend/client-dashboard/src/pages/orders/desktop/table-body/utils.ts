import { TableCellSizeEnum } from 'phoqer';

export interface HeaderType {
    key: string;
    title: string;
}

export const headerOrder: HeaderType[] = [
    { key: 'status', title: 'Status' },
    { key: 'options', title: 'Options' },
    { key: 'images', title: 'Images' },
    { key: 'title', title: 'Title' },
    { key: 'price', title: 'Price' },
    { key: 'user', title: 'Author' },
    { key: 'address', title: 'Address' },
    { key: 'comment', title: 'Comment' },
    { key: 'date', title: 'Created' },
    { key: 'expired', title: 'Expiration Date' },
];

export const sizeMap: { [key: string]: TableCellSizeEnum } = {
    status: TableCellSizeEnum.Small,
    options: TableCellSizeEnum.Middle,
    title: TableCellSizeEnum.Large3Extra,
    images: TableCellSizeEnum.Small2Extra,
    price: TableCellSizeEnum.Middle,
    address: TableCellSizeEnum.Large4Extra,
    comment: TableCellSizeEnum.Large3Extra,
    user: TableCellSizeEnum.LargeExtra,
    date: TableCellSizeEnum.Large2Extra,
    expired: TableCellSizeEnum.Large2Extra,
};
