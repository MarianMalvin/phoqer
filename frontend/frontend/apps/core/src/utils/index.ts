export * from './recentOffers';
export * from './errorHandler';
