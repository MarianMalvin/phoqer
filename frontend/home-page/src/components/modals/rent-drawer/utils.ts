import { RentPayload } from './type';

export const RENT_DRAWER_DATA_KEY = 'rent-drawer-data';

export const initialValues: RentPayload = {
    country: '',
    city: '',
    zip: '',
    address: '',
    comment: '',
};

export const getInitialValues = (): RentPayload => {
    try {
        const json = localStorage.getItem(RENT_DRAWER_DATA_KEY);
        if (!json) {
            return initialValues;
        }

        const data = JSON.parse(json);
        return data || initialValues;
    } catch {
        return initialValues;
    }
};

export const setInitialValues = (value: RentPayload): void => {
    try {
        localStorage.setItem(RENT_DRAWER_DATA_KEY, JSON.stringify(value));
    } catch (error) {
        console.log(error);
    }
};
