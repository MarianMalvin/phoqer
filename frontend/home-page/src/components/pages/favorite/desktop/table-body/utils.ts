export interface HeaderType {
    key: string;
    title: string;
}

export const headerOrder: HeaderType[] = [
    { key: 'image', title: 'Image' },
    { key: 'options', title: 'Rent now' },
    { key: 'title', title: 'Title' },
    { key: 'price', title: 'Price' },
    { key: 'sale', title: 'Sale' },
    { key: 'category', title: 'Category' },
];

export const sizeMap: { [key: string]: number } = {
    image: 10,
    title: 45,
    price: 15,
    sale: 10,
    category: 12,
    options: 18,
};
