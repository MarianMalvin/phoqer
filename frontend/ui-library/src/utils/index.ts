export * from './copy';
export * from './css-property';
export * from './get-css-class';
export * from './label-size-map';
