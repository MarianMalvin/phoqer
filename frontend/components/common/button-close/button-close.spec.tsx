import { fireEvent, screen } from '@testing-library/react';

import { render } from '../../../tests/tests.utils';
import ButtonClose from './index';

describe('Test ButtonClose component', () => {
    it('Should match snapshot', () => {
        const { asFragment } = render(<ButtonClose />, {});
        expect(asFragment()).toMatchSnapshot();
    });

    it('Should match snapshot', () => {
        const { asFragment } = render(<ButtonClose className="test" type="submit" />, {});
        expect(asFragment()).toMatchSnapshot();
    });

    it('Test onClick', () => {
        const handleClick = jest.fn();
        render(<ButtonClose onClick={handleClick} />, {});
        fireEvent.click(screen.getByRole(/button/i));
        expect(handleClick).toHaveBeenCalledTimes(1);
    });
});
