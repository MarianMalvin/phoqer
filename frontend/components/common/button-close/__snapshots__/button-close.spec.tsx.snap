// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test ButtonClose component Should match snapshot 1`] = `
<DocumentFragment>
  <button
    class="btn-0-2-1"
    type="button"
  />
</DocumentFragment>
`;

exports[`Test ButtonClose component Should match snapshot 2`] = `
<DocumentFragment>
  <button
    class="btn-0-2-2 test"
    type="submit"
  />
</DocumentFragment>
`;
